src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs

```diff
@@ -47,5 +48,137 @@ internal static (VirtualNetworkStream ClientStream, VirtualNetworkStream ServerS
+            using (RSA server = RSA.Create())
+            {
+                CertificateRequest rootReq = new CertificateRequest(
+                    "CN=Root",
+                    root,
+                    HashAlgorithmName.SHA256,
+                    RSASignaturePadding.Pkcs1);
+
+                rootReq.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(true, false, 0, true));
+                rootReq.CertificateExtensions.Add(
+                    new X509SubjectKeyIdentifierExtension(rootReq.PublicKey, false));
+                rootReq.CertificateExtensions.Add(
+                    new X509KeyUsageExtension(X509KeyUsageFlags.KeyCertSign, false));
+
+                //DateTimeOffset start = DateTimeOffset.UtcNow.AddMinutes(-5);
+                DateTimeOffset endTime = startTime.AddMonths(1);
+
+                X509Certificate2 rootCertWithKey = rootReq.CreateSelfSigned(startTime, endTime);
+
+                CertificateRequest intermedReq = new CertificateRequest(
+                   "CN=Intermediate",
+                   intermediate,
+                   HashAlgorithmName.SHA256,
+                   RSASignaturePadding.Pkcs1);
+
+                intermedReq.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(true, false, 0, true));
+                intermedReq.CertificateExtensions.Add(
+                    new X509SubjectKeyIdentifierExtension(intermedReq.PublicKey, false));
+                intermedReq.CertificateExtensions.Add(
+                    new X509KeyUsageExtension(X509KeyUsageFlags.KeyCertSign, false));
+
+                byte[] serial = new byte[8];
+                RandomNumberGenerator.Fill(serial);
+
+                X509Certificate2 intermedCertWithKey;
+                using (X509Certificate2 intermedPub = intermedReq.Create(rootCertWithKey, startTime, endTime, serial))
+                {
+                    intermedCertWithKey = intermedPub.CopyWithPrivateKey(intermediate);
+                }
+
+                CertificateRequest serverReq = new CertificateRequest(
+                     $"CN={name}",
+                     server,
+                     HashAlgorithmName.SHA256,
+                     RSASignaturePadding.Pkcs1);
+
+                serverReq.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(false, false, 0, false));
+                serverReq.CertificateExtensions.Add(
+                    new X509SubjectKeyIdentifierExtension(serverReq.PublicKey, false));
+
+                // Add Issuer KeyIdentifier
+                using (SHA1 sha1 = SHA1.Create())
+                {
+                    byte[] data = new byte[24];
+                    data[0] = 0x30; //SEQUENCE
+                    data[1] = 22;
+                    data[2] = 0x80;
+                    data[3] = 20;
+                    Buffer.BlockCopy(sha1.ComputeHash(intermedCertWithKey.PublicKey.EncodedKeyValue.RawData), 0, data, 4, 20);
```

Comment:
You should copy this from the subject key identifier value, not re-compute it. https://github.com/dotnet/runtime/blob/531ebcd1895aa32a106db5849cd4ace3909cefce/src/libraries/System.Security.Cryptography.X509Certificates/tests/RevocationTests/CertificateAuthority.cs#L735-L783

Reply:
If I read this correctly, this would copy the KeyID from parent certificate, right? 
But that does not seems right as the format seems to be slightly different. 
(openssl x509 -text shows garbage with the simple copy)

Reply:
It is a different structure, but it copies it into the correct place within the structure.